### PR TITLE
Feature/91 position od

### PR DESCRIPTION
**Pull request by thibaudteil**
* **Tickets addressed:** bsk-091
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This tickets adds a SRuKF filter to process full camera position vectors. It is nearly an exact replica of the flyby filter, motivating the generation of an interface.

## Verification
As with other filters, the test checks for propagation and measurements. The post fit residuals are seen to reconverge into the measurement noise when the dynamics shift.

## Documentation
RST documentation

## Future work
Refactor part of the interface

### CODE REVIEW

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp]: :large_blue_circle: Could throw this back inline above.

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp]: :large_orange_diamond: typo in `Residuals`

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp]: :large_orange_diamond: Are the implied semantics of `computePostFits`, post fits are available/have been computed (given the time tag from the measured data is new compared to previous step), I think that's right? I see the header file comment.

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.h
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.h]: :large_orange_diamond: Let's quickly touch base about adjusting some of these to `const` references and removing the const on some parameters if we truly want them copied in to the function.

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp
Comment (patkenneally) [inline src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp:1]: :large_blue_circle: It looks like SourceTree is adding some text to the commit messages based on them being cherry picked?

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp
Comment (thibaudteil) [reply src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp]: Done

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp
Comment (thibaudteil) [reply src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp]: Done

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp
Comment (thibaudteil) [reply src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp]: changed it to measurementRead to be more consistent, and placed it in the readMeasurement function

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp
Comment (thibaudteil) [reply src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.cpp:1]: Fixed

File: src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.h
Comment (thibaudteil) [reply src/fswAlgorithms/opticalNavigation/positionODuKF/positionODuKF.h]: Fixed

**Pull request merged**
